WordPress/wporg-gp-translation-events
# Email notifications

This PR adds email notifications to all the attendees of an event around 1 hour and 1 day before the event.

To test this PR, I recommend you this configuration:
- [WP Crontrol](https://wordpress.org/plugins/wp-crontrol/) plugin to view the scheduled jobs.
- [Mailtrap](https://mailtrap.io/) or some similar tool to be able to easily review the emails sent by this PR.
- [WP Mail SMTP](https://es.wordpress.org/plugins/wp-mail-smtp/) or another SMTP plugin to easily configure the email sending to Mailtrap.

To test this PR, you can:

**Create an event**. In this example, I created an event for the May month. As I am locally work in CEST timezone (UTC+2) and the form is using UTC, I need to start on 2024-04-30 22:00 (2024-05-01 00:00 UTC time) and finish on 2024-05-31 21:59 (2024-05-31 23:59 UTC time).

![image](https://github.com/WordPress/wporg-gp-translation-events/assets/1667814/3dd5aad2-2840-4cf9-af93-d0d009bff5d0)

Once I published the event, with the **WP Crontrol** plugin I can see the custom cron events (in CEST timezone):
- One 1 hour before the event: 2024-04-30 23:00 CEST.
- One 23 hours before the event: 2024-04-30 00:00 CEST.

You can see in the notifications:
- The `post_id` (event_id) as parameter.
- The `Wporg\TranslationEvents\Notifications\Notifications_Send->send_notifications()` as the action that will be executed when this event will be fired.

![image](https://github.com/WordPress/wporg-gp-translation-events/assets/1667814/d9f3bef4-57a0-49c9-8444-ca96d0743609)

Each time I modified the event, all these custom cron events are deleted and re-scheduled, to use the new start time.

To fire one of these events manually, you can use the `Run now` link, available in each custom cron event.

![image](https://github.com/WordPress/wporg-gp-translation-events/assets/1667814/1e21bc4b-affc-43b5-9b23-8db8cb3da2c8)

Once the custom cron event is fired, you will receive an email like this:

![image](https://github.com/WordPress/wporg-gp-translation-events/assets/1667814/a33231ad-ac52-448b-8f8f-1e6af9f9a2f9)

Fixes https://github.com/WordPress/wporg-gp-translation-events/issues/168.
